mews-operations/availabilityBlockProcessor.md

```diff
@@ -0,0 +1,176 @@
+
+
+#### Dates
+| Property | Type | Contract | Description                                                                                                                                                                          |
+| :-- | :-- | :-- |:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `start` | `string` | required | Start date in format `yyyy-mm-dd`, e.g. `2024-12-05`; this is the first day of the block and so can be booked for stay.                                                              |
+| `end` | `string` | required | End date in format `yyyy-mm-dd`, e.g. `2024-12-08`; this day is _not_ included in the block and so cannot be booked for stay (it is analogous to reservation departure).             |
+
+> **Dates**: Interval for 1 night (e.g. is represented 2024-12-24/25 is represented as `"start": "2024-12-24", "end": "2024-12-24"`).
+
+
+#### Release strategy
+| Property | Type | Contract | Description                                                                                  |
+| :-- | :-- | :-- |:---------------------------------------------------------------------------------------------|
+| `type` | `string` | required | release strategy for the availability block, in the first version only support "fixed" |
+| `fixedDate` | `string` | required | release date in format `"yyyy-MM-dd"` \(e.g., `"2024-12-08"`\).                              |
+
+
+#### Space category allocation
+| Property | Type | Contract | Description                                              |
+| :-- | :-- |:---------|:---------------------------------------------------------|
+| `spaceTypeCode` | `string` | required | Space type code of the block.               |
+| `allocatedSpaces` | [`Space count`](#space-count) collection | required | The number of spaces **allocated** in the time interval. |
+| `rates` | [`Rates`](#rate) collection | optional | Prices for the space category.                   |
+
+
+### Space count
+| Property | Type | Contract | Description                                                              |
+| :-- | :-- | :-- |:-------------------------------------------------------------------------|
+| `start` | `string` | required | Start date in format `"yyyy-MM-dd"` \(e.g. `"2024-12-05"`\).             |
+| `end` | `string` | required | End date \(excluded\) in format `"yyyy-MM-dd"` \(e.g., `"2024-12-08"`\). |
+| `count` | `int` | required | Number of spaces (base of usage either occupied or allocated.)           
+
+> **Dates**: Interval for 1 night (e.g. is represented 2024-12-24/25 is represented as `"start": "2024-12-24", "end": "2024-12-24"`).
```

Comment:
technically we can do this as interval but it would mean extend little the API layer to decompose it